central/metrics/custom/tracker/aggregator.go

```diff
@@ -0,0 +1,78 @@
+package tracker
+
+import (
+	"iter"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+type aggregationKey string // e.g. IMPORTANT_VULNERABILITY_SEVERITY|true
+
+// aggregatedRecord is a single gauge metric record.
+type aggregatedRecord struct {
+	labels prometheus.Labels
+	total  int
+}
+
+// aggregator computes the aggregation result.
```

Comment:
Could you please provide a more concrete example here as a comment? This comment just tells the obvious (the aggregator aggregates...).

The reason I think that examples are especially important here, is because this is a struct with complicated and nested types.